Update networking section of components overview (#7689)
kubevirt/kubevirt: docs/components.md

## Networking

KubeVirt aims to seamlessly integrate into Kubernetes networking, with VMs
connected to the same network as Pods, using the same resources and same APIs.
Without any external extensions, VMs in KubeVirt are capable of using the pod
network, Services, or NetworkPolicies.

To implement this, VMs are only bound to network resources that are made
available to the virt-launcher Pod's network namespace. With this model, a wide
variety of networking plugins can be supported.

> We also care for advanced networking scenarios, such as use of service meshes,

Reviewer: ```suggestion
We also care for advanced networking scenarios, such as connecting to service meshes,
```
(I found it hard to parse the original)

Author: Works for me. Fixed
